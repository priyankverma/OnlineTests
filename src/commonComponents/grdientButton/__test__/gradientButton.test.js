import React from "react";
import ReactDOM from "react-dom";
import GradientButton from "../gradientButton";
import { render } from "@testing-library/react";
import renderer from "react-test-renderer";

it("renders without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(<GradientButton></GradientButton>, div);
});

it("renders correctly", () => {
  const { getByTestId } = render(
    <GradientButton title="Continue"></GradientButton>
  );
  expect(getByTestId("button")).toHaveTextContent("Continue");
});

it("renders correctly", () => {
  const { getByTestId } = render(
    <GradientButton title="Proceed"></GradientButton>
  );
  expect(getByTestId("button")).toHaveTextContent("Proceed");
});

it("Matches Snapshot Continue", () => {
  const tree = renderer
    .create(<GradientButton title="Continue"></GradientButton>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
it("Matches Snapshot Proceed", () => {
  const tree = renderer
    .create(<GradientButton title="Proceed"></GradientButton>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
