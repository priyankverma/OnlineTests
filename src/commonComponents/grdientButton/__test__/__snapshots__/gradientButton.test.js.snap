// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches Snapshot Continue 1`] = `
<div
  className="sc-bdvvaa ksCvjJ"
  data-testid="button"
  height="15px"
  onClick={[Function]}
  width="30%"
>
  <b>
    Continue
  </b>
</div>
`;

exports[`Matches Snapshot Proceed 1`] = `
<div
  className="sc-bdvvaa ksCvjJ"
  data-testid="button"
  height="15px"
  onClick={[Function]}
  width="30%"
>
  <b>
    Proceed
  </b>
</div>
`;
